From 9cf20568bbd03d1283f4bdbc2a2f10b10246841f Mon Sep 17 00:00:00 2001
From: Patrick Delaunay <patrick.delaunay@foss.st.com>
Date: Fri, 19 Jan 2024 10:56:17 +0100
Subject: [PATCH 548/954] feat(stm32mp25-fdts): add DDR power supplies

Add the required power supplies for DDR nodes. The power supplies are
provided by STPMIC2 regulators.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 0b27ecd293e9bda59b6c52a52e021c7b4ae4db6e feat(stm32mp25-fdts): add DDR power supplies]
Change-Id: I3f604ab9f93a0ff66190575924d47dc881f2e0a3
Signed-off-by: Patrick Delaunay <patrick.delaunay@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/355237
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/368532
Reviewed-by: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 fdts/stm32mp257f-dk.dts  | 6 ++++++
 fdts/stm32mp257f-ev1.dts | 7 +++++++
 2 files changed, 13 insertions(+)

diff --git a/fdts/stm32mp257f-dk.dts b/fdts/stm32mp257f-dk.dts
index d6e140c3c6..51caf708be 100644
--- a/fdts/stm32mp257f-dk.dts
+++ b/fdts/stm32mp257f-dk.dts
@@ -47,6 +47,12 @@
 	};
 };
 
+&ddr {
+	vdd1-supply = <&vdd1_ddr>;
+	vdd2-supply = <&vdd2_ddr>;
+	vddq-supply = <&vdd2_ddr>;
+};
+
 &hash {
 	status = "okay";
 };
diff --git a/fdts/stm32mp257f-ev1.dts b/fdts/stm32mp257f-ev1.dts
index 0e6e8d3afc..848ab89925 100644
--- a/fdts/stm32mp257f-ev1.dts
+++ b/fdts/stm32mp257f-ev1.dts
@@ -47,6 +47,13 @@
 	};
 };
 
+&ddr {
+	vdd-supply = <&vdd_ddr>;
+	vtt-supply = <&vtt_ddr>;
+	vpp-supply = <&vpp_ddr>;
+	vref-supply = <&vref_ddr>;
+};
+
 &hash {
 	status = "okay";
 };
-- 
2.53.0

